### ISSUE TITLE
[BUG] Harvester Not Showing As Cloud Provider With KDM Edit From Rancher v2.7.0

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#3389]
**Describe the bug**
Harvester doesn't display as an available cloud provider when editing the KDM on Rancher of:
https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json

From Rancher v2.7.0 logs, KDM Data gets populated:
```
2023/01/19 22:39:21 [INFO] Refreshing driverMetadata in 1 minutes
2023/01/19 22:39:21 [INFO] getReleaseConfig: reloading config for k3s
2023/01/19 22:39:21 [INFO] getReleaseConfig: reloading config for rke2
2023/01/19 22:39:22 [INFO] driverMetadata: refreshing data from upstream https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json
2023/01/19 22:39:22 [INFO] Retrieve data.json from local path /var/lib/rancher-data/driver-metadata/data.json
2023/01/19 22:39:22 [INFO] Loaded configuration from https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json in [0x6cddfb8]
2023/01/19 22:39:22 [INFO] Loaded configuration from https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json in [0x6cddfb8]
```

**To Reproduce**
1. Have an available Rancher v2.7.0 instance available, import Harvester
1. Modify Rancher's Global-Settings, KDM Metadata, update URL to: https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json
1. Navigate to Cluster Management -> Drivers -> click call-to-action "Refresh KDM Metadata" button
1. Attempt to provision RKE2 with Harvester & Harvester Cloud Credential
1. Cloud provider for "Harvester" is greyed out.
1. Note that, if KDM-Metadata gets updated to "dev-v2.7", via URL: https://releases.rancher.com/kontainer-driver-metadata/dev-v2.7/data.json , same process of Refreshing KDM Metadata takes place, RKE2 can provision on Harvester as Harvester is not "greyed out" in Cloud Provider selection

**Expected behavior**
Harvester to be an available cloud provider.

**Environment**
 - Harvester ISO version:
     - v1.1.1
     - master-14bbee2c-head
 - Underlying Infrastructure qemu/kvm

**Additional context**
Related Issue:
https://github.com/rancher/rke2/issues/3765

Screenshots:

![Screenshot from 2023-01-19 14-41-58](https://user-images.githubusercontent.com/5370752/213771654-6ce98aa2-1cf4-4220-a67c-73d8b38cbbfe.png)
![Screenshot from 2023-01-19 14-35-34](https://user-images.githubusercontent.com/5370752/213771656-6af9f34c-5c6a-4b44-9514-75ec99eae0b3.png)
![Screenshot from 2023-01-19 14-29-51](https://user-images.githubusercontent.com/5370752/213771658-49a8eb86-ac05-42e8-bc6d-e202d71e51b1.png)

**Comment by WuJun2016** [comment]
@irishgordo   The cause of this problem is that the version of the harvester-cloud-provider or harvester-csi-provider does not match. 
case1: use `https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json`
1. rke2 chart `harvester-cloud-provider`:  0.1.14
`harvester-cloud-provider` Specifies the range of compatible matches：  >=0.0.1 <0.1.14
`0.1.14` No satisfied `>=0.0.1 <0.1.14`
2. csi version: `0.1.15`  ,  range:  `>=0.0.1 <0.1.15`,  so `0.1.15` No satisfied `>=0.0.1 <0.1.15`

case2: use `https://releases.rancher.com/kontainer-driver-metadata/release-v2.7/data.json`
1. rke2 version:  `0.1.13`, range: `>=0.0.1 <0.1.14` ,  `0.1.13` satisfied `>=0.0.1 <0.1.14`
2. csi version: `0.1.11`, range: `>=0.0.1 <0.1.15`,     `0.1.11` satisfied  `>=0.0.1 <0.1.15`

So this is in line with expectations

**Comment by irishgordo** [comment]
@WuJun2016 thanks for the response back :smile:  - that makes sense that the conditionals in place are causing that.
For, https://github.com/rancher/rke2/issues/3765 - working through testing it - there was the expectation that: https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json - would be used.
It was mentioned that perhaps the constraints that are in place needed to be adjusted?
https://github.com/rancher/rke2/issues/3765#issuecomment-1398997123 , like: `0.1.1400`

**Comment by WuJun2016** [comment]
> @WuJun2016 thanks for the response back 😄 - that makes sense that the conditionals in place are causing that. For, [rancher/rke2#3765](https://github.com/rancher/rke2/issues/3765) - working through testing it - there was the expectation that: https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json - would be used. It was mentioned that perhaps the constraints that are in place needed to be adjusted? [rancher/rke2#3765 (comment)](https://github.com/rancher/rke2/issues/3765#issuecomment-1398997123) , like: `0.1.1400`

@irishgordo  To fix this problem, we need two steps：
1. When the Harvester version is less than v1.2, we do not judge the compatibility (because we cannot change the setting configuration of the released Harvester)，  Completed and submitted to racher/dashboard repo
2. Modify the configuration of Harvester (create a new issue, which will be fixed in v1.2)  ： https://github.com/harvester/harvester/issues/3404

**Comment by WuJun2016** [comment]
@irishgordo  
Note: 
The same problem exists in rancher 2.6-head, so validation is also needed。  thanks

**Comment by guangbochen** [comment]
Will need to add a release note as the 2023-jan KDM release will set the Harvester cloud provider unusable; users need to bump to the newer Rancher version of v2.6.11 or v2.7.2.

**Comment by TachunLin** [comment]
Verified **fixed** on Rancher `v2.6-head` and `v2.7-head` with Harvester v1.1.2 (master-aa7fcafc-head)
Close this issue.

### Result
After we update the rke-metadata-config URL to January 2023 rke2 
https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json

The Harvester can show as cloud provider in RKE2 cluster provisioning page. 

* v2.7-head (61dae8d)
![image](https://user-images.githubusercontent.com/29251855/217455717-e142ac90-7285-4ac6-859a-2c1d419fd73e.png)

* v2.6-head (815c94c)
![image](https://user-images.githubusercontent.com/29251855/217461276-2f252dfe-d602-4666-a63a-8580cf000fc7.png)

### Test Information
* Test Environment: 1 nodes harvester on provo bare machine
* Harvester version: v1.1.2 (master-aa7fcafc-head)
* Rancher version: v2.7-head (61dae8d), v2.6-head (815c94c)

### Verify Steps
1. Prepare Rancher v2.7-head and v2.6-head
2. Edit rke-metadata-config  in Rancher global settings, set the url to https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/january_2023_rke2/data/data.json
3. Import Harvester and create cloud credential
4. Provision RKE2 cluster
5. Check the cloud provider drop down list
